### PR TITLE
Test on Rails 5.2.0.rc1

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by tegon** [comment]
Hello @jcoyne, thanks for your pull request.
Could you do a rebase with master, please?

File: test/rails_app/config/application.rb [hunk 1]
```diff
@@ -31,8 +31,10 @@ class Application < Rails::Application
     # config.assets.enabled = false
 
     config.action_mailer.default_url_options = { host: "localhost", port: 3000 }
-
-    if DEVISE_ORM == :active_record && (Rails::VERSION::MAJOR >= 4 && Rails::VERSION::MINOR >= 2)
+    rails_version = Gem::Version.new(Rails.version)
```
Comment (tegon) [inline test/rails_app/config/application.rb]: Would you mind explaining why do we need this change?

Comment (jcoyne) [reply test/rails_app/config/application.rb]: `raise_in_transactional_callbacks` was deprecated in rails 5.0 and removed in 5.2 https://github.com/rails/rails/commit/8029f779b8a1dd9848fee0b7967c2e0849bf6e07

**Comment by jcoyne** [comment]
@tegon I've rebased this on master.

File: test/rails_app/config/application.rb (hunk 1 above)
Comment (tegon) [reply test/rails_app/config/application.rb]: I see. The strange thing is that some specs are failing on CI with an error that seems to be related to this
```
`method_missing': undefined method `raise_in_transactional_callbacks=' for ActiveRecord::Base:Class
```
Could you check those errors?

**Comment by jcoyne** [comment]
@tegon I've got this passing in its entirety except for the mongoid builds.  I have no idea about what's going on with mongoid. Can anyone offer help?

File: test/rails_app/config/application.rb (hunk 1 above)
Comment (jcoyne) [reply test/rails_app/config/application.rb]: This is now fixed.

**Comment by tegon** [comment]
@jcoyne It seems that the problem is because you didn't add the `mongoid` gem to your `Gemfile`.

**Comment by jcoyne** [comment]
@tegon I see, I copied the rails 5.0 gemfile, which also has mongoid commented out, so I added an exclusion to the matrix as rails 5.0 does.

File: test/rails_app/config/application.rb (hunk 1 above)
Comment (gregoirenovel) [reply test/rails_app/config/application.rb]: @jcoyne is it fixed on `master`? I still encounter this error

**Comment by tegon** [comment]
@jcoyne You're right, I don't know why it's excluding `mongoid` but since it's already doing this, I'll take a look at this later.
The build still isn't passing, though. Can you take a look, please?

**Comment by jcoyne** [comment]
@tegon i have looked, I don't know what that is about... it seems to be something related to how the testing framework is stubbing rails?

**Comment by tegon** [comment]
I'm lost on this one too. @rafaelfranca do you know what might be causing this error?

**Comment by jasnow** [comment]
Look at this: https://github.com/rails/rails/pull/29699

**Comment by rafaelfranca** [comment]
`ActiveRecord::Migrator` don't have a `migrate` method anymore.

We need to change this

https://github.com/plataformatec/devise/blob/master/test/orm/active_record.rb#L7

to:

`MigrationContext.new(File.expand_path("../../rails_app/db/migrate/", __FILE__).migrate`

**Comment by tegon** [comment]
@jcoyne Can you make this change, please?
I think we'll have to keep both syntaxes with a branch condition so that it works on multiple Rails versions.

**Comment by jcoyne** [comment]
@tegon Yes, I'll get to this presently.

**Comment by jcoyne** [comment]
@tegon that seems to have fixed the problem

@rafaelfranca Thank you for your assistance.

**Pull request merged**